.changeset/fluffy-donuts-end.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+Migrate plugin to use the new auth services.
+**BREAKING**: Add a required service discovery to the router options and remove the optional identity from it. Also change the permissions type to be `PermissionsService`.
```

Comment:
We can't have a breaking change in the scaffolder atm. Let's do what we did with catalog and default to `HostDiscovery` if `discovery` is not provided.

Reply:
Done